src/Lean/MetavarContext.lean

```diff
@@ -620,6 +620,21 @@ def instantiateMVarsCore (mctx : MetavarContext) (e : Expr) : Expr × MetavarCon
     instantiateExprMVars e
   runST fun _ => instantiate e |>.run |>.run mctx
 
+/-
+Substitutes assigned metavariables in `e` with their assigned value according to the
+`MetavarContext`, recursively.
+
+Example:
+```
+run_meta do
+  let mvar1 ← mkFreshExprMVar (mkConst `Nat)
+  let e := (mkConst `Nat.succ).app mvar1
+  logInfo e -- Nat.succ ?m.773
+  mvar1.mvarId!.assign (mkNatLit 42)
+  let e' ← instantiateMVars e
+  logInfo e' -- Nat.succ 42
```

Comment:
I like this example, but it doesn't quite demonstrate what it intends to. If you move the first `logInfo` down one line, you will still get `Nat.succ 42` as the result, because `MessageData` automatically follows instantiated metavariables. To see the effect of `instantiateMVars` you need to use one of the lower level expr printing methods which does not follow instantiations.